### PR TITLE
Feature make card rank and suit immutable

### PR DESCRIPTION
**Pull request by salwil**
Task: #190

### CODE REVIEW

File: src/main/java/ch/uzh/ifi/hase/soprafs22/screwyourneighborserver/entity/Card.java [hunk 1]
```diff
@@ -7,11 +7,14 @@
 import javax.persistence.*;
 
 @Entity
+@Table(uniqueConstraints = {@UniqueConstraint(columnNames = {"hand_id", "round_id"})})
```
Comment (BacLuc) [inline src/main/java/ch/uzh/ifi/hase/soprafs22/screwyourneighborserver/entity/Card.java:10]: For later: is isnt it possible to use the java property names?

File: src/main/java/ch/uzh/ifi/hase/soprafs22/screwyourneighborserver/web/GlobalExceptionHandler.java [hunk 2]
```diff
@@ -25,6 +25,12 @@ private ResponseEntity<String> handleConstraintViolation(ConstraintViolationExce
     return ResponseEntity.badRequest().body(e.getMessage());
   }
 
+  @ExceptionHandler(value = {org.springframework.dao.DataIntegrityViolationException.class})
```
Comment (BacLuc) [inline src/main/java/ch/uzh/ifi/hase/soprafs22/screwyourneighborserver/web/GlobalExceptionHandler.java:28]: An import statement for DataIntegrityViolationException would be nice

File: src/test/java/ch/uzh/ifi/hase/soprafs22/screwyourneighborserver/integration/CardConstraintIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  void change_cards_value_does_not_work() {
+    Card card = new Card(CardRank.QUEEN, CardSuit.HEART);
+    cardRepository.saveAll(List.of(card));
+    Collection<Card> cardsBeforeSave = cardRepository.findAll();
+    card.setCardRank(CardRank.KING);
+    cardRepository.saveAll(List.of(card));
+    Collection<Card> cardsAfterSave = cardRepository.findAll();
+
+    assertEquals(1, cardsBeforeSave.size());
+    assertEquals(CardRank.QUEEN, cardsBeforeSave.stream().findAny().get().getCardRank());
```
Comment (BacLuc) [inline src/test/java/ch/uzh/ifi/hase/soprafs22/screwyourneighborserver/integration/CardConstraintIntegrationTest.java:82]: cardsBeforeSave.stream().findAny().get() could be extracted into a variable

File: src/test/java/ch/uzh/ifi/hase/soprafs22/screwyourneighborserver/integration/CardConstraintIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    assertEquals(CardSuit.HEART, cardsBeforeSave.stream().findAny().get().getCardSuit());
+    assertEquals(1, cardsAfterSave.size());
+    assertEquals(CardRank.QUEEN, cardsAfterSave.stream().findAny().get().getCardRank());
```
Comment (BacLuc) [inline src/test/java/ch/uzh/ifi/hase/soprafs22/screwyourneighborserver/integration/CardConstraintIntegrationTest.java:85]: cardsAfterSave.stream().findAny().get() could be extracted into a variable

**Review by BacLuc — APPROVED** [review]

File: src/main/java/ch/uzh/ifi/hase/soprafs22/screwyourneighborserver/entity/Card.java (hunk 1 above)
Comment (salwil) [reply src/main/java/ch/uzh/ifi/hase/soprafs22/screwyourneighborserver/entity/Card.java:10]: You mean hand and round? No unfortunately not.

**Review by BacLuc — APPROVED** [review]

**Pull request merged**